### PR TITLE
 Fix CI Java build using correct Autotools env var (closes #1883) 

### PR DESCRIPTION
**Pull request by DaAwesomeP**
This supersedes #1883 and solves the issue found in #1881.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -90,17 +90,22 @@ jobs:
...
       - name: Set additional Linux configure arguments
         if: runner.os == 'Linux'
         # Silence all deprecated declarations on Linux due to auto_ptr making the build log too long
         run: |
-          echo "GH_OLA_CONFIGURE_ARGS=$GH_OLA_CONFIGURE_ARGS CPPFLAGS=-Wno-deprecated-declarations" >> $GITHUB_ENV
+          echo "DISTCHECK_CONFIGURE_FLAGS=$DISTCHECK_CONFIGURE_FLAGS CPPFLAGS=-Wno-deprecated-declarations" >> $GITHUB_ENV
+      - name: Set additional Linux distcheck configure arguments
+        if: matrix.task == 'distcheck'
+        run: |
+          echo "DISTCHECK_CONFIGURE_FLAGS=${{ matrix.configure-args }} CPPFLAGS=-Wno-deprecated-declarations" >> $GITHUB_ENV
```
Comment (peternewman) [inline .github/workflows/build.yaml:104]: I think this is redundant from my shonky testing isn't it?
```suggestion
```

**Review by peternewman — CHANGES_REQUESTED** [review]
Do you want to try and fix the codespell typos too?

**Comment by peternewman** [comment]
The distcheck is failing, it looks because this isn't in 0.10, so it might be easiest to just skip testing the Java stuff on 0.10, but the new distcheck stuff can then go into master too, where Java should build, or might need the Java updates from #1882 :
https://github.com/OpenLightingProject/ola/pull/702/files

**Comment by DaAwesomeP** [comment]
> Do you want to try and fix the codespell typos too?

See #1886.

> The distcheck is failing, it looks because this isn't in 0.10, so it might be easiest to just skip testing the Java stuff on 0.10, but the new distcheck stuff can then go into master too, where Java should build, or might need the Java updates from #1882 : https://github.com/OpenLightingProject/ola/pull/702/files

OK, I modified this to fix the distcheck stuff but disable Java for now. Your following pull can re-enable it.

**Comment by DaAwesomeP** [comment]
@peternewman polite ping!

**Review by peternewman — APPROVED** [review]
LGTM thanks!

**Pull request merged**